src/Pact/Compile.hs

```diff
@@ -404,7 +489,12 @@ letsForm = do
 useForm :: Compile (Term Name)
 useForm = do
   modName <- (_atomAtom <$> userAtom) <|> str <|> expected "bare atom, string, symbol"
-  TUse (ModuleName modName) <$> optional hash' <*> contextInfo
+  i <- contextInfo
+  u <- Use (ModuleName modName) <$> optional hash' <*> pure i
+  -- this is the one place module may not be present, use traversal
+  psUser . csModule . _Just . msImports %= (u:)
+  return $ TUse u i
```

Comment:
If we're going down this "ModuleState" route, we could do the same for many of the other constructs are aren't necessarily terms, where we're just aggregating names. Why couldn't we also do what we did to `TImplements` to `TUse` as well? we could add a `msUses :: [Use]` to `ModuleState`, and skip a few steps throughout the evaluation process.